AK/FloatingPoint.h

```diff
@@ -12,6 +12,57 @@
 
 namespace AK {
 
+template<typename T>
+union FloatExtractor;
+
+#if ARCH(I386) || ARCH(X86_64) || ARCH(AARCH64)
+// This assumes long double is 80 bits, which is true with GCC on Intel platforms
```

Comment:
I know it didn't have this before but maybe we can add a `static_assert(sizeof(long double) == 10)` here to validate what this comment is saying?

Reply:
FYI, `long double` is the same as `double` (64 bits) in the Apple arm64 ABI.

Reply:
sizeof(long double) is 12 in -m32 mode with both gcc and clang. so seems an x86 32-bitism. on 64-bit both compilers report 16.

https://godbolt.org/z/Ps7TWsj3j

Reply:
So any assumptions about IEEE 754 quad floats vs Intel 8087 "Extended Precision" 80-bit floats should probably be verified by static_asserts, yes. As-is, the comment is incorrect in the face of the preprocessor `#if`

Reply:
It seems that `sizeof` and `bit_cast<long double>` do not work, so I ended up checking for the value of `__LDBL_MAX__`. (Yes, I'm aware of NumericLimits but don't think it is rational to include it for one compile-time check)

Reply:
Eh. It will only be 80-bit long double on i686 and such, or x86_64 platforms where you used -mlong-double-80. Otherwise it will "probably" be 16 bit IEEE 754 quad-float. ... Except on apple platforms like Daniel mentioned, where double and long double are the same thing. 

So I think to resolve this we should update the check by removing the current static assert, and updating the comment to say "FIXME: assumes long double is Intel x87 80-bit floating point, which is only true on some x86 platforms" and leave it for later

Reply:
We shouldn't be using long double in any serenity code at the moment other than the userspace emulator, so this being not quite right will only affect ports.

Reply:
Really? https://godbolt.org/z/T8Kx1fG6r

Reply:
Since we are not targeting MSVC, we can use __LDBL_MANT_DIG__ and __LDBL_MAX_EXP__ to be absolutely sure.
https://godbolt.org/z/7TxrYbqWx